tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -2788,30 +2788,18 @@ void EnqueueAllocateBufferImpl(AllocBufferMetadata alloc_md) {
     uint32_t allocated_addr;
     if (is_sharded(buffer->buffer_layout())) {
```

Comment:
Can we move allocation from this `if` statement? 
Seems like the only difference is that one allocates `buffer->size()` size and the other `buffer->num_dev_pages() * buffer->page_size()`

Also.. its probably best to move this logic inside the buffer itself

Reply:
I left it like this for now as I recently discussed with @TT-BrianLiu about the issues regarding this and how we want to change it. The reason it's like this is because the `buffer->size()` should reflect the on device size, but for sharded buffers it does not and actually reflects the unpadded/host size, and we have dependencies related to sharding infra that seems to expect it being this way for now. I could add a `buffer::device_size()` api if we want to avoid the branching here, but left it like this since I was expecting Brian to clean it up with his planned changes.